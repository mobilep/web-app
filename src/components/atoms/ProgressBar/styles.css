.ProgressBar {
	position: relative;

	/* Number */
	--numberLineHeight: 17px;
	--numberFontSize: 12px;

	/* Progress */
	--progressHeight: 2px;
	--progressBorderRadius: 1px;
}

.ProgressBar-bar {
	height: var(--progressHeight);
	border-radius: var(--progressBorderRadius);
	background-color: var(--tomatoRed);
	opacity: .28;
}

.ProgressBar-value {
	position: absolute;
	left: 0;
	top: var(--numberLineHeight);
	height: var(--progressHeight);
	border-radius: var(--progressBorderRadius);
	background-color: var(--tomatoRed);
}

.ProgressBar-number {
	font-size: var(--numberFontSize);
	line-height: var(--numberLineHeight);
	color: var(--grey3);
	font-weight: var(--normal);
	text-align: right;
}
