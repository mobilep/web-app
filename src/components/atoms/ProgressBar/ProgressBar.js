import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './styles.css';

const getClassName = (classname) => classNames('ProgressBar', { [classname]: classname });

const ProgressBar = ({ value, className }) => (
	<div className={getClassName(className)}>
		<div className='ProgressBar-number'>{value} %</div>
		<div className='ProgressBar-bar' />
		<div className='ProgressBar-value' style={{ width: `${value}%` }} />
	</div>
);

ProgressBar.propTypes = {
	className: PropTypes.string,
	value: PropTypes.number,
};

export default ProgressBar;
