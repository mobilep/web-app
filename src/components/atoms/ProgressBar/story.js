import React from 'react';
import { storiesOf } from '@storybook/react';
import ProgressBar from './ProgressBar';

const props = {
	value: 16,
};

storiesOf('1. Atoms / ProgressBar', module)
	.add('ProgressBar', () => (
		<ProgressBar { ...props } />
	));
