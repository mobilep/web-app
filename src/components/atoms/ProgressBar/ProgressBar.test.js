import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from './ProgressBar';

describe('Atom - ProgressBar', () => {

	it('Should render', () => {
		const wrapper = shallow(<ProgressBar value={15} />);
		expect(wrapper).toMatchSnapshot();
	});

});
