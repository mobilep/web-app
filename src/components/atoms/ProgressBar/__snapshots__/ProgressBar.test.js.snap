// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - ProgressBar Should render 1`] = `
<div
  className="ProgressBar"
>
  <div
    className="ProgressBar-number"
  >
    15
     %
  </div>
  <div
    className="ProgressBar-bar"
  />
  <div
    className="ProgressBar-value"
    style={
      Object {
        "width": "15%",
      }
    }
  />
</div>
`;
